app/javascript/lib/metrics-api.js

```diff
@@ -0,0 +1,277 @@
+  const numFilters =
+    Object.values(facets).reduce((prevNumFilters, filterArray) => {
+      return prevNumFilters + filterArray.length
+    }, 0)
+
+  return [numFacets, numFilters]
+}
+
+
+/**
+ * Returns human-friendly list of applied filters for each facet
+ *
+ * This enables us to more feasibly answer "What filters are people using?"
+ *
+ * Renames keys in facets for easier discoverability as event properties in
+ * Mixpanel.  E.g. instead of "disease" and "species", which will not appear
+ * together in Mixpanel's alphabetized list, log these as "filtersDisease" and
+ * "filtersSpecies".
+ *
+ * Also renames filters from opaque IDs (e.g. MONDO_0018076) to human-readable
+ * labels (e.g. tuberculosis).
+ */
+function getFriendlyFilterListByFacet(facets) {
+  const filterListByFacet = {}
+  Object.entries(facets).forEach(([facet, filters]) => {
+    const friendlyFacet = `filters${facet[0].toUpperCase() + facet.slice(1)}`
+    const friendlyFilters = filters.map(filterId => {
+      // This global variable is initialized in application.html.erb
+      // and populated in scp-api.js
+      return window.SCP.filterNamesById[filterId]
+    })
+    filterListByFacet[friendlyFacet] = friendlyFilters
+  })
+  return filterListByFacet
+}
+
+/**
+ * Log study search metrics.  Might support gene, cell search in future.
+ */
+export function logSearch(type, terms, facets, page) {
+  if (isPageLoadSearch === true) {
+    // This prevents over-reporting searches.
+    // Loading home page triggers search, which is a side-effect / artifact
+    // with regard to tracking user interactions.  This variable is set to
+    // false once per page load as a way to omit such artifacts from logging.
+    isPageLoadSearch = false
+    return
+  }
+
+  const numTerms = getNumberOfTerms(terms)
+  const [numFacets, numFilters] = getNumFacetsAndFilters(facets)
+  const facetList = Object.keys(facets)
+
+  const filterListByFacet = getFriendlyFilterListByFacet(facets)
+
+  const simpleProps = {
+    type, terms, page,
+    numTerms, numFacets, numFilters, facetList
+  }
+  const props = Object.assign(simpleProps, filterListByFacet)
+
+  log('search', props)
+
+  // Google Analytics fallback: remove once Bard and Mixpanel are ready for SCP
+  ga( // eslint-disable-line no-undef
+    'send', 'event', 'faceted-search', 'study-search',
+    'num-terms', numTerms
+  )
+}
+
+/**
+ * Log filter search metrics
+ */
+export function logFilterSearch(facet, terms) {
+  const numTerms = getNumberOfTerms(terms)
+
+  const defaultProps = { facet, terms }
+  const props = Object.assign(defaultProps, { numTerms })
+  log('search-filter', props)
+
+  // Google Analytics fallback: remove once Bard and Mixpanel are ready for SCP
+  ga( // eslint-disable-line no-undef
+    'send', 'event', 'faceted-search', 'search-filter',
+    'num-terms', numTerms
+  )
+}
+
+/**
+ * Log when a download is authorized.
+ * This is our best web-client-side methodology for measuring downloads.
+ */
```

Comment:
likewise, this function should live alongside the download code. (DownloadButton.js?)

Reply:
See [comment](https://github.com/broadinstitute/single_cell_portal_core/pull/454#discussion_r393842055).